Add VM mount to backup workspaces and prefs on host file system

eivantsov opened the pull request:
@TylerJewell @riuvshin please take a look. This way we auto mount a current dir into `~/che-storage` which is configured as storage for Che workspaces and prefs inside a VM. As a result, even when a box is destroyed, workspaces, projects and prefs are there.

Signed-off-by: Eugene Ivantsov <eivantsov@codenvy.com>

riuvshin commented:
LGBT

TylerJewell commented:
Three basic changes proposed.  

1.  Let's name the directory /home/vagrant/.che.
2.  Let's copy /conf/che.properties to .che/che.properties.
3.  Let's update the properties in that file instead of appending.

riuvshin commented:
+1 on this

eivantsov commented:
@TylerJewell @riuvshin I have updated the PR. che.properties is copied and updated now in ~/.che which is set to be the location of CHE_LOCAL_CONF_DIR

TylerJewell commented:
@eivantsov LGTM.  Question - did you not need to source .bashrc after you modify the file?  I am surprised that the modification to the environment variable was picked up.

eivantsov commented:
It's picked up since I both export it in the command line and in .bashrc. So, it works for Vagrant because of export in the command line, and for user (when logged in) in .bashrc.

TylerJewell commented:
Great - will you consider doing a similar PR for the ARTIK IDE Vagrantfile?

eivantsov commented:
yes, sure
